src/app/inventory/store/override-sockets.ts

```diff
@@ -93,12 +95,45 @@ export function applySocketOverrides(
     };
   });
 
+  const overriddenSockets = {
+    ...item.sockets,
+    allSockets,
+  };
+
+  // If the item is a subclass with sockets, calculate the active fragments from aspects and remove
+  // any that are either disabled or
+  if (item.bucket.hash === BucketHashes.Subclass) {
+    const aspects = getSocketsByCategoryHash(overriddenSockets, SocketCategoryHashes.Aspects);
+    let activeFragments = 0;
+    for (const aspect of aspects) {
+      activeFragments += aspect.plugged?.plugDef.plug.energyCapacity?.capacityValue || 0;
+    }
+
+    const fragments = getSocketsByCategoryHash(overriddenSockets, SocketCategoryHashes.Fragments);
+    for (const fragment of fragments) {
+      // if the fragment is active and we have an override for it, we leave it along as it has already
+      // been overridden
+      if (activeFragments > 0 && socketOverrides[fragment.socketIndex]) {
+        activeFragments--;
```

Comment:
I'm not actually sure if the fragment count should be enforced in this function. This is often used in UI code to show the fully configured item and I'm worried about a situation where UI code silently hides fragments when the user hasn't selected aspects (yet).

Reply:
One of the issues that I noticed was we were including stats from fragments that were inactive on the item, think 3 fragment slots but 4 fragments are socketed. Maybe we can still show it, but grey it out and not include the stats like the game does.

Reply:
Having a look at it, I might need to introduce a new property to the socket to indicate if the plug is active/inactive. There is an `enabled` property on the `DimPlug` we [cache those too](https://github.com/DestinyItemManager/DIM/blob/master/src/app/inventory/store/sockets.ts#L663).